swaScraper.py

```diff
@@ -5,6 +5,128 @@
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support import expected_conditions as EC
 
+def correct_date(date):
+    """
+    Converts the date format to one accepted by SWA
+
+    SWA form cannot accept slashes for dates and is in the format YYYY-MM-DD
+    :param date: Date string to correct
+    :return: Corrected date string
+
+    """
+    a, b, c = date.split("/")
+
+    if(len(a) == 4):
+        # Assumed format is year, month, day
+        return "%s-%s-%s"%(a, b, c)
+    else:
+        # Assumed format is month, day, year
+        return "%s-%s-%s" % (c, a, b)
+
+def correct_time_string(timeString):
+    """
+    Returns a valid string version of the time of departure/arrival based on
+    given argument value
+
+    If no-value is given, "ALL_DAY" will be returned
+
+    :param timeString: String to correct
+    :return: Corrected string value
+    """
+    if timeString == "BEFORE_NOON":
+        return "BEFORE_NOON"
+
+    elif timeString == "NOON_TO_6PM":
+        return "NOON_TO_SIX"
+
+    elif timeString == "AFTER_6PM":
+        return "AFTER_SIX"
+
+    else:
+        return "ALL_DAY"
+
+def direct_load(args, browser):
+    """
+    Creates and sends a HTTP GET request for ticket prices based on the
+    given arguments
+
+    :param args: Arguments used to form request
+    :param browser: Browser object used to make request
+    :return: None
+    """
+    
+    commandString = ""
+
+    if args.passengers:
+        # Set number of passengers.
+        commandString += "adultPassengersCount=%s&" % (args.passengers)
+    else:
+        commandString += "adultPassengersCount=%s&" % ("0")
+
+    # Set departure date.
+    cleanedUpDate = correct_date(args.departure_date)
+    commandString += "departureDate=%s&" % (cleanedUpDate)
+
+    # Set the departure time
+    correctedTime = correct_time_string(args.departure_time)
+    commandString += "departureTimeOfDay=%s&" % (correctedTime)
+
+    # Set the arrival airport.
+    commandString += "destinationAirportCode=%s&" % (args.arrive)
+
+    # How is the fare to be paid, USD or POINTS
+    commandString += "fareType=USD&"
+
+    commandString +="int=HOMEQBOMAIR&leapfrogRequest=true&"
+
+    # Set the departing airport.
+    commandString += "originationAirportCode=%s&" % (args.depart)
+
+    # If there are any seniors load the senior column
+    # If there are no seniors, bnut the senior column is loaded, no prices
+    # will get loaded
+    if args.seniors and int(args.seniors) > 0:
+        commandString += "passengerType=SENIOR&"
+    else:
+        commandString += "passengerType=ADULT&"
+
+    # Promo code entry
+    commandString += "promoCode=&"
+
+    # No idea what this is for, but it is in the GET string
+    commandString += "redirectToVision=true&reset=true&"
```

Comment:
Pretty sure this can be removed.

Reply:
It looks like there are no changes in the results if the fields `redirectToVision` and `reset` are left out, but I put them in there since they are in GET request when you fill out the form normally.